packages/flutter_tools/lib/src/commands/run.dart

```diff
@@ -288,11 +292,32 @@ class RunCommand extends RunCommandBase {
       );
     }
 
+    DateTime appStartedTime;
+    // Sync completer so the completing agent attaching to the resident doesn't 
+    // need to know about analytics. 
+    //
+    // Do not add more operations to the future.
+    final Completer<Null> appStartedTimeRecorder = new Completer<Null>.sync();
+    appStartedTimeRecorder.future.then(
+      (Null _) { appStartedTime = clock.now(); }
```

Comment:
I'm confused as to what this is trying to do. Wouldn't `scheduleMicrotask` do the same thing cleanly? Or `Future.sync`?